src/file.ts

```diff
@@ -337,6 +364,13 @@ class RequestError extends Error {
   errors?: Error[];
 }
 
+type ValueOf<T> = T[keyof T];
+type HeaderValue = ValueOf<http.OutgoingHttpHeaders>;
+
+const SEVEN_DAYS = 7 * 24 * 60 * 60;
+
+const DEFAULT_SIGNING_VERSION = 'v2';
+
 /**
  * A File object is created from your {@link Bucket} object using
  * {@link Bucket#file}.
```

Comment:
Is it possible to create a signed URL to list objects in a bucket e.g, GET request without an object name? Is that something to do in a follow-up PR?

Reply:
I believe we've discussed this and concluded that this is out of scope for this PR. Adding bucket level signed URLs will be a future endeavour.